src/components/Form/index.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+
+import type {
+    Comment,
+    CommentReply,
+} from '../../state/comments';
+
+
+export interface CommentFormSetProps {
+    comments: Comment[];
+}
+
+export function CommentFormSetComponent(props: CommentFormSetProps) {
+    const [initialNumber] = React.useState(props.comments.filter(comment => {return comment.remoteId != null}).length);
+    const prefix = "comments";
+
+    const commentForms = props.comments.map((comment, formNumber) => (
+        <CommentFormComponent
+            key={comment.localId}
+            comment={comment}
+            formNumber={formNumber}
+        />
+    ));
+
+    return (<>
+        <PrefixedHiddenInput
+            fieldName="TOTAL_FORMS"
+            value={props.comments.length}
+            prefix={prefix}
+        />
+        <PrefixedHiddenInput
+            fieldName="INITIAL_FORMS"
+            value={initialNumber}
+            prefix={prefix}
+        />
+        <PrefixedHiddenInput
+            fieldName="MIN_NUM_FORMS"
+            value="0"
+            prefix={prefix}
+        />
+        <PrefixedHiddenInput
+            fieldName="MAX_NUM_FORMS"
+            value=""
+            prefix={prefix}
+        />
+        {commentForms}
+    </>)
+}
+
+interface PrefixedHiddenInputProps {
+    prefix: string;
+    value: number|string;
+    fieldName: string;
+}
+
+function PrefixedHiddenInput(props: PrefixedHiddenInputProps) {
+    return (<input
+        type="hidden"
+        name={`${props.prefix}-${props.fieldName}`}
+        value={props.value} 
+        id={`id_${props.prefix}-${props.fieldName}`}
+    />)
+}
+
+export interface CommentReplyFormComponentProps {
+    reply: CommentReply;
+    prefix: string;
+    formNumber: number;
+}
+
+export function CommentReplyFormComponent(props: CommentReplyFormComponentProps) {
+    const reply = props.reply
```

Comment:
**Suggestion:** Exactly for this reason you should just destruct things from the props.

```tsx
function a({ reply, ...props })
```

Reply:
Good point, done this now.